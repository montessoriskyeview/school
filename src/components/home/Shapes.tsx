import "./styles.css";
import { useRef, useLayoutEffect } from "react";
import { degreesToRadians, mix } from "popmotion";
import { Object3D } from "three";

const colors = [
  "#FFFFFF",
  "#03fcfc",
  "#fc037f",
  "#fcc203",
  "#0b03fc",
  "#9403fc",
  "#fc7303",
  "#fc0303",
  "#03fc4e",
];

function getRandomColor() {
  // Return the element at the random index
  return colors[Math.floor(Math.random() * colors.length)];
}


export const Icosahedron = () => (
  <mesh rotation-x={0.35}>
    <icosahedronGeometry args={[1, 0]} />
    <meshBasicMaterial wireframe color={getRandomColor()} />
  </mesh>
);

export const Star = ({ p }: { p: number }) => {
  const ref = useRef<Object3D>(null);

  useLayoutEffect(() => {
    const distance = mix(2, 3.5, Math.random());
    const yAngle = mix(
      degreesToRadians(80),
      degreesToRadians(100),
      Math.random()
    );
    const xAngle = degreesToRadians(360) * p;
    ref.current!.position.setFromSphericalCoords(distance, yAngle, xAngle);
  });

  return (
    <mesh ref={ref as any}>
      <boxGeometry args={[0.05, 0.05, 0.05]} />
      <meshBasicMaterial wireframe color={getRandomColor()} />
    </mesh>
  );
};