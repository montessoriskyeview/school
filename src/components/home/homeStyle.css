.box {
	padding: 160px 20px;
	display: grid;
	gap: 250px;
	background-color: transparent;
	top: 0;
	left: 0;

	@media only screen and (max-width: 600px) {
		padding: 100px 0px;
	}
}

.content {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	min-height: 100vh;
	/* Set minimum height for full viewport */
}

.info-text {
	background-color: #00000077;
	padding: 20px;
	margin: 50px;
	border-radius: 10px;

	@media only screen and (max-width: 600px) {
		padding: 10px;
		margin: 10px;
	}
}

.map-info {
	background-color: #00000077;
	padding: 20px;
	margin: 50px;
	border-radius: 10px;
	display: flex;
	flex-direction: column;

	@media only screen and (max-width: 600px) {
		padding: 10px;
		margin: 10px;
	}
}

.map-container {
	width: 50%;
	height: 300px;

	align-self: center;

	@media only screen and (max-width: 600px) {
		width: 90%;
	}
}