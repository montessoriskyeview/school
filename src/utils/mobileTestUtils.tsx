import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Mobile viewport configurations
export const MOBILE_VIEWPORTS = {
  iPhoneSE: { width: 375, height: 667 },
  iPhone12: { width: 390, height: 844 },
  iPhone12ProMax: { width: 428, height: 926 },
  SamsungGalaxyS20: { width: 360, height: 800 },
  iPad: { width: 768, height: 1024 },
};

// Set mobile viewport for testing
export const setMobileViewport = (
  viewport: keyof typeof MOBILE_VIEWPORTS = 'iPhoneSE'
) => {
  const { width, height } = MOBILE_VIEWPORTS[viewport];

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  Object.defineProperty(window, 'innerHeight', {
    writable: true,
    configurable: true,
    value: height,
  });

  // Trigger resize event
  window.dispatchEvent(new Event('resize'));
};

// Custom render function with mobile context
const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// Mobile-specific test helpers
export const mobileTestUtils = {
  // Test touch target size
  expectTouchTarget: (element: HTMLElement, minSize = 44) => {
    const rect = element.getBoundingClientRect();
    expect(rect.width).toBeGreaterThanOrEqual(minSize);
    expect(rect.height).toBeGreaterThanOrEqual(minSize);
  },

  // Test mobile spacing
  expectMobileSpacing: (element: HTMLElement, minSpacing = 24) => {
    const computedStyle = window.getComputedStyle(element);
    const padding =
      parseFloat(computedStyle.paddingTop) +
      parseFloat(computedStyle.paddingBottom);
    const margin =
      parseFloat(computedStyle.marginTop) +
      parseFloat(computedStyle.marginBottom);
    const totalSpacing = padding + margin;
    expect(totalSpacing).toBeGreaterThanOrEqual(minSpacing);
  },

  // Test mobile accessibility
  expectMobileAccessibility: (element: HTMLElement) => {
    const hasRole =
      element.getAttribute('role') || element.tagName.toLowerCase();
    const hasLabel =
      element.getAttribute('aria-label') ||
      element.getAttribute('alt') ||
      element.textContent?.trim();

    expect(hasRole).toBeTruthy();
    expect(hasLabel).toBeTruthy();
  },

  // Test mobile responsive behavior
  expectMobileResponsive: (element: HTMLElement) => {
    const computedStyle = window.getComputedStyle(element);
    const fontSize = parseFloat(computedStyle.fontSize);
    const lineHeight = parseFloat(computedStyle.lineHeight);

    // Check minimum font size for mobile
    expect(fontSize).toBeGreaterThanOrEqual(16);

    // Check minimum line height for readability
    expect(lineHeight).toBeGreaterThanOrEqual(1.4);
  },

  // Test mobile navigation
  expectMobileNavigation: (element: HTMLElement) => {
    const isClickable =
      element.tagName === 'BUTTON' ||
      element.tagName === 'A' ||
      element.getAttribute('role') === 'button' ||
      element.onclick !== null;

    expect(isClickable).toBe(true);
  },
};

// Export custom render
export { customRender as render };

// Re-export everything from testing library
export * from '@testing-library/react';
