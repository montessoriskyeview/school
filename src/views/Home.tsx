import { InfoText } from "../components/shared/InfoText";
import { CanvasView } from "../components/shared/Canvas/CanvasView";

export const Home = () => {
  return (
    <CanvasView>
      <InfoText
        title="Montessori School Skye View"
        text="A Private Chez Nous School"
        titleVariant="h3"
      />
      <InfoText
        title="About us"
        text="Montessori Skye View is a new Montessori school dedicated to fostering a love of learning in children aged 3 and up. We believe in the power of the Montessori Method, an educational approach developed by Dr. Maria Montessori, which emphasizes self-directed exploration, hands-on activities, and a nurturing environment. Located in the northwest corner of Las Vegas, our beautiful Skye Canyon campus provides ample outdoor space for play and learning."
      />
    </CanvasView>
  )
}