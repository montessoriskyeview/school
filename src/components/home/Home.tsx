import { Box, Button, Snackbar, Tooltip } from "@mui/material";

import "./homeStyle.css";
import React from "react";

import LinkIcon from '@mui/icons-material/Link';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { ContentContainer } from "../shared/ContentContainer";
import { Typography } from "../shared/Typography";

export const MapInfo = ({ title }: { title: string }) => {
	const defaultProps = {
		center: {
			lat: 36.2893875,
			lng: -115.311701
		},
		zoom: 15
	};

	const [open, setOpen] = React.useState(false);

  const link = `https://maps.google.com/?q=${defaultProps.center.lat},${defaultProps.center.lng}&z=${defaultProps.zoom}`;

  const onClickCopy = async () => {
    try {
      await navigator.clipboard.writeText(link);
      setOpen(true);
    } catch (error) {
      console.error('Failed to copy to clipboard:', error);
      // Fallback: try to copy using document.execCommand
      try {
        const textArea = document.createElement('textarea');
        textArea.value = link;
        document.body.appendChild(textArea);
        textArea.select();
        document.execCommand('copy');
        document.body.removeChild(textArea);
        setOpen(true);
      } catch (fallbackError) {
        console.error('Fallback copy also failed:', fallbackError);
      }
    }
  };

  const onClickLink = () => {
    try {
      window.open(link, '_blank');
    } catch (error) {
      console.error('Failed to open link:', error);
    }
  };

	const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
		if (reason === 'clickaway') {
      return;
    }
		setOpen(false);
	};

	return (
		<ContentContainer
			sx={{
				alignContent: "center"
			}}
		>
			<Typography variant="h2"
				sx={{
					paddingBottom: "10px",
					color: "white",
				}}>
				{title}
			</Typography>
			<Box component="div"
				sx={{
					width: "50%",
					alignSelf: "center",
					display: "flex",
					justifyContent: "center",
				}}
			>
				<Tooltip
					title="Open in new tab"
				>
					<Button
						onClick={onClickLink}
					>
						<OpenInNewIcon />
					</Button>
				</Tooltip>
				<Tooltip title="Copy link">
					<Button
						onClick={onClickCopy}
					>
						<LinkIcon />
					</Button>
				</Tooltip>
			</Box>
			<Snackbar
				open={open}
				autoHideDuration={5000}
				onClose={handleClose}
				message="Copied!"
			/>
		</ContentContainer >
	)
}
