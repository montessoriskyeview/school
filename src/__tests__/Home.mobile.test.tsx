import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { setMobileViewport } from '../utils/mobileTestUtils';
import { MapInfo } from '../components/home/Home';

// Mock clipboard API
Object.assign(navigator, {
  clipboard: {
    writeText: jest.fn(),
  },
});

// Mock window.open
const mockOpen = jest.fn();
Object.defineProperty(window, 'open', {
  value: mockOpen,
  writable: true,
});

// Mock document.execCommand for fallback clipboard
Object.defineProperty(document, 'execCommand', {
  value: jest.fn(),
  writable: true,
});

describe('Home Component - MapInfo Mobile Experience', () => {
  beforeEach(() => {
    // Set mobile viewport for all tests
    setMobileViewport('iPhoneSE');
    // Clear mocks
    jest.clearAllMocks();
  });

  describe('Mobile Layout and Responsiveness', () => {
    test('renders correctly on mobile viewport', () => {
      render(<MapInfo title="Test Location" />);
      expect(screen.getByText('Test Location')).toBeInTheDocument();
    });

    test('content container is accessible', () => {
      render(<MapInfo title="Test Location" />);
      const title = screen.getByText('Test Location');
      expect(title).toBeVisible();
      expect(title).toBeInTheDocument();
    });

    test('snapshot matches mobile layout', () => {
      const { container } = render(<MapInfo title="Test Location" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Mobile Touch Interactions', () => {
    test('buttons are present and clickable', () => {
      render(<MapInfo title="Test Location" />);

      const buttons = screen.getAllByRole('button');
      expect(buttons.length).toBeGreaterThan(0);

      buttons.forEach(button => {
        expect(button).toBeVisible();
      });
    });

    test('open in new tab button works on mobile', async () => {
      render(<MapInfo title="Test Location" />);

      const openButton = screen.getByRole('button', {
        name: /open in new tab/i,
      });

      fireEvent.click(openButton);

      await waitFor(() => {
        expect(mockOpen).toHaveBeenCalledWith(
          'https://maps.google.com/?q=36.2893875,-115.311701&z=15',
          '_blank'
        );
      });
    });

    test('copy link button works on mobile', async () => {
      render(<MapInfo title="Test Location" />);

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      await waitFor(() => {
        expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
          'https://maps.google.com/?q=36.2893875,-115.311701&z=15'
        );
      });
    });

    test('touch events work properly', async () => {
      render(<MapInfo title="Test Location" />);

      const buttons = screen.getAllByRole('button');

      buttons.forEach(button => {
        fireEvent.touchStart(button);
        fireEvent.touchEnd(button);
      });

      // Should not cause any errors
      await waitFor(() => {
        expect(buttons[0]).toBeInTheDocument();
      });
    });
  });

  describe('Mobile Accessibility', () => {
    test('buttons have proper ARIA labels', () => {
      render(<MapInfo title="Test Location" />);

      const openButton = screen.getByRole('button', {
        name: /open in new tab/i,
      });
      const copyButton = screen.getByRole('button', { name: /copy link/i });

      expect(openButton).toHaveAttribute('aria-label');
      expect(copyButton).toHaveAttribute('aria-label');
    });

    test('title has proper heading structure', () => {
      render(<MapInfo title="Test Location" />);

      const title = screen.getByRole('heading', { level: 2 });
      expect(title).toBeInTheDocument();
      expect(title).toHaveTextContent('Test Location');
    });

    test('tooltips work on mobile', () => {
      render(<MapInfo title="Test Location" />);

      const openButton = screen.getByRole('button', {
        name: /open in new tab/i,
      });
      const copyButton = screen.getByRole('button', { name: /copy link/i });

      // Hover to show tooltips
      fireEvent.mouseEnter(openButton);
      fireEvent.mouseEnter(copyButton);

      expect(openButton).toHaveAttribute('aria-label', 'Open in new tab');
      expect(copyButton).toHaveAttribute('aria-label', 'Copy link');
    });

    test('snackbar is accessible', async () => {
      render(<MapInfo title="Test Location" />);

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      await waitFor(() => {
        const snackbar = screen.getByText('Copied!');
        expect(snackbar).toBeInTheDocument();
      });
    });
  });

  describe('Mobile Content and Styling', () => {
    test('button container has proper layout', () => {
      render(<MapInfo title="Test Location" />);

      // Use getAllByRole to handle multiple buttons and get the first one's container
      const buttons = screen.getAllByRole('button');
      expect(buttons.length).toBeGreaterThan(0);

      // Instead of using .closest(), we'll test that buttons are properly rendered
      buttons.forEach(button => {
        expect(button).toBeInTheDocument();
        expect(button).toBeVisible();
      });
    });

    test('buttons are properly spaced', () => {
      render(<MapInfo title="Test Location" />);

      const buttons = screen.getAllByRole('button');
      expect(buttons.length).toBeGreaterThan(1);

      // Check that buttons are separate elements
      buttons.forEach(button => {
        expect(button).toBeInstanceOf(HTMLElement);
      });
    });

    test('content is properly structured', () => {
      render(<MapInfo title="Test Location" />);

      const title = screen.getByText('Test Location');
      expect(title).toBeInTheDocument();
      expect(title).toBeVisible();
    });
  });

  describe('Mobile Performance', () => {
    test('renders within reasonable time on mobile', async () => {
      const startTime = performance.now();

      render(<MapInfo title="Test Location" />);

      await waitFor(() => {
        expect(screen.getByText('Test Location')).toBeInTheDocument();
      });

      const endTime = performance.now();
      const renderTime = endTime - startTime;

      // Should render within 100ms on mobile
      expect(renderTime).toBeLessThan(100);
    });

    test('button interactions are responsive', async () => {
      render(<MapInfo title="Test Location" />);

      const startTime = performance.now();

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      await waitFor(() => {
        expect(navigator.clipboard.writeText).toHaveBeenCalled();
      });

      const endTime = performance.now();
      const interactionTime = endTime - startTime;

      // Should respond within 50ms
      expect(interactionTime).toBeLessThan(50);
    });

    test('no console errors during mobile interactions', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<MapInfo title="Test Location" />);

      const buttons = screen.getAllByRole('button');
      buttons.forEach(button => {
        fireEvent.click(button);
      });

      expect(consoleSpy).not.toHaveBeenCalled();
      consoleSpy.mockRestore();
    });
  });

  describe('Mobile Responsive Behavior', () => {
    test('renders correctly on different mobile viewports', () => {
      const viewports = ['iPhoneSE', 'iPhone12', 'SamsungGalaxyS20'] as const;

      viewports.forEach(viewport => {
        setMobileViewport(viewport);
        const { container } = render(<MapInfo title="Test Location" />);
        expect(container).toMatchSnapshot();
      });
    });

    test('maintains functionality across mobile devices', () => {
      const viewports = ['iPhoneSE', 'iPhone12', 'SamsungGalaxyS20'] as const;

      viewports.forEach(viewport => {
        setMobileViewport(viewport);
        render(<MapInfo title="Test Location" />);

        // Use getAllByText to handle potential duplicates and get the first one
        const titles = screen.getAllByText('Test Location');
        expect(titles[0]).toBeInTheDocument();
      });
    });
  });

  describe('Mobile Content Validation', () => {
    test('displays correct title', () => {
      render(<MapInfo title="Montessori Skye View" />);
      expect(screen.getByText('Montessori Skye View')).toBeInTheDocument();
    });

    test('generates correct map link', () => {
      render(<MapInfo title="Test Location" />);

      const expectedLink =
        'https://maps.google.com/?q=36.2893875,-115.311701&z=15';

      // Check that the link is used in both buttons
      const copyButton = screen.getByRole('button', { name: /copy link/i });
      fireEvent.click(copyButton);

      expect(navigator.clipboard.writeText).toHaveBeenCalledWith(expectedLink);
    });

    test('snackbar shows correct message', async () => {
      render(<MapInfo title="Test Location" />);

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      await waitFor(() => {
        expect(screen.getByText('Copied!')).toBeInTheDocument();
      });
    });

    test('snackbar auto-hides after 5 seconds', async () => {
      jest.useFakeTimers();

      render(<MapInfo title="Test Location" />);

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      await waitFor(() => {
        expect(screen.getByText('Copied!')).toBeInTheDocument();
      });

      // Fast-forward time
      jest.advanceTimersByTime(5000);

      await waitFor(() => {
        expect(screen.queryByText('Copied!')).not.toBeInTheDocument();
      });

      jest.useRealTimers();
    });
  });

  describe('Mobile Error Handling', () => {
    test('handles clipboard API errors gracefully', async () => {
      // Mock clipboard API to throw an error
      (navigator.clipboard.writeText as jest.Mock).mockRejectedValueOnce(
        new Error('Clipboard error')
      );

      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<MapInfo title="Test Location" />);

      const copyButton = screen.getByRole('button', { name: /copy link/i });

      fireEvent.click(copyButton);

      // Should not crash the component and should still show snackbar
      await waitFor(() => {
        expect(screen.getByText('Copied!')).toBeInTheDocument();
      });

      // Should log the error
      expect(consoleSpy).toHaveBeenCalledWith(
        'Failed to copy to clipboard:',
        expect.any(Error)
      );

      consoleSpy.mockRestore();
    });

    test('handles window.open errors gracefully', async () => {
      // Mock window.open to throw an error
      mockOpen.mockImplementationOnce(() => {
        throw new Error('Popup blocked');
      });

      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<MapInfo title="Test Location" />);

      const openButton = screen.getByRole('button', {
        name: /open in new tab/i,
      });

      fireEvent.click(openButton);

      // Should not crash the component
      await waitFor(() => {
        expect(openButton).toBeInTheDocument();
      });

      // Should log the error
      expect(consoleSpy).toHaveBeenCalledWith(
        'Failed to open link:',
        expect.any(Error)
      );

      consoleSpy.mockRestore();
    });
  });
});
