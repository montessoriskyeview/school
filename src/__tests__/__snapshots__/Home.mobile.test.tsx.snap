// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component - MapInfo Mobile Experience Mobile Layout and Responsiveness snapshot matches mobile layout 1`] = `
<div>
  <div
    class="MuiBox-root css-jvnuv8"
    spacing="md"
    variant="default"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-gtpl0f-MuiTypography-root"
    >
      Test Location
    </h2>
    <div
      class="MuiBox-root css-1yzwf4e"
    >
      <button
        aria-label="Open in new tab"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Copy link"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Component - MapInfo Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 1`] = `
<div>
  <div
    class="MuiBox-root css-jvnuv8"
    spacing="md"
    variant="default"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-gtpl0f-MuiTypography-root"
    >
      Test Location
    </h2>
    <div
      class="MuiBox-root css-1yzwf4e"
    >
      <button
        aria-label="Open in new tab"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Copy link"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Component - MapInfo Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 2`] = `
<div>
  <div
    class="MuiBox-root css-jvnuv8"
    spacing="md"
    variant="default"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-gtpl0f-MuiTypography-root"
    >
      Test Location
    </h2>
    <div
      class="MuiBox-root css-1yzwf4e"
    >
      <button
        aria-label="Open in new tab"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Copy link"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Component - MapInfo Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 3`] = `
<div>
  <div
    class="MuiBox-root css-jvnuv8"
    spacing="md"
    variant="default"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-gtpl0f-MuiTypography-root"
    >
      Test Location
    </h2>
    <div
      class="MuiBox-root css-1yzwf4e"
    >
      <button
        aria-label="Open in new tab"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Copy link"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
